### DISCUSSION TITLE
[Accordion] Is there a way to force accordion content height re-calculation after new DOM is added?

### DISCUSSION THREAD

**Discussion by aslauris** [radix-ui/primitives#2562] [Help]
I've got a dynamic form inside `<Accordion.Content />` where new rows/items could be added. However, `<Accordion.Content />` height is not updated after new DOM elements are added resulting in new rows being invisible. Is there any way to force accordion content height re-calculation or any other workaround to this? Many thanks!

**Comment by benoitgrelard** [comment]
I believe this is something you would have to deal with yourself in this specific case.

**Comment by dschlabach** [comment]
Hey! Find a solution for this? @aslauris

**Reply by Fabricevladimir** [reply]
Did you? 🙈

**Reply by dschlabach** [reply]
No 😭 , ended up doing something different entirely

**Reply by Fabricevladimir** [reply]
I ended up passing `forceMount` to the content. It works...most of time!

**Reply by aslauris** [reply]
Hi @dschlabach sorry for delay - haven't noticed your question. Actually, I ended up with a solution with no accordion as well ;)

**Comment by bipinmv** [comment]
Haven't got a solution yet? Here is a workaround for you.

The CSS variable --radix-accordion-content-height doesn't get updated when the elements inside AccordionContent changes.
We need to add a resize observer to update the value.

``` typescript
import * as Accordion from '@radix-ui/react-accordion';

export const Content = ({ children}: { children: ReactNode }) => {
  const ref = useRef(null);
  const contentRef = useRef(null);

  // Radix Accordion doesnt update height when children updates.
  // We need to manualy add a resize observer on children of the Content component
  // to update the CSS variable height of the content.

  useEffect(() => {
    const content = contentRef.current;

    if (!ref.current || !content) return;

    const resizeObserver = new ResizeObserver(() => {
      const currentHeight = ref.current.clientHeight;

      content.style.cssText = `--radix-accordion-content-height: ${currentHeight}px;`;
    });

    resizeObserver.observe(ref.current);

    return () => resizeObserver.disconnect(); // clean up
  }, []);

  return (
    <Accordion.Content ref={contentRef}>
      <div ref={ref}>{children}</div>
    </Accordion.Content>
  );
};
```

**Reply by devrefframe** [reply]
Thank you!!

**Reply by seeniolabode** [reply]
solid
